src/internal_events/splunk_hec.rs

```diff
@@ -160,6 +160,16 @@ mod sink {
             decrement_gauge!("splunk_pending_acks", self.count);
         }
     }
+
+    pub struct SplunkEventTimestampWarning<'a> {
+        pub message: &'a str,
+    }
+
+    impl<'a> InternalEvent for SplunkEventTimestampWarning<'a> {
+        fn emit(self) {
+            warn!(message = %self.message);
+        }
+    }
```

Comment:
👍 to pulling this out into an internal event.

We typically like the events structured rather than templating messages though. I think this could look something like two different events with two different `emit` implementations with separate static messages:

```rust
pub type SplunkEventTimestampMissing;

pub struct SplunkEventTimestampInvalidType {
  r#type: String # this would be used in the emitted warning as trace label like `warn!(message = "Timestamp was an unexpected type", type=self.r#type)
}
```

In general we also prefer to use log fields rather than templating log messages. This makes it easier to, for example, generate metrics from them.

We should also add `internal_log_rate_secs = 10` to the `warn!` call. Log messages in Vector (aside from `TRACE` messages) are generally rate limited to avoid flooding the user's logs (and disk) in the case of catastrophic errors. See other examples in this file.

I'm ok with not publishing metrics here yet since we are still trying to define which metrics should be published for "warning" events.

Reply:
That makes sense... thanks for the details! I've refined the internal events in the latest commit.